docs_src/design/adr/core/0026-Core-Data-Cache.md

```diff
@@ -0,0 +1,85 @@
+            - this operation would be called as part of the core data add event/reading operations
+            - this option causes core data to keep within a size range (between a max and min number of readings) 
+            - examples: 
+                - if there are 100 readings in the database and the `CacheSize` of the database readings is set to 100, then when the next reading is added to core data, a delete of the latest `DeleteSize` readings are removed (starting with the oldest readings) from core data.
+                - if there are 80 readings in the database and the `CacheSize` of the database readings is set to 100, then when the next reading is added to core data, no delete occurs.  The reading gets added normally and the number of readings in the database is now 81.
+            - this option reduces the add/delete thrashing.  It only deletes when the max size is hit or exceeded
+
+!!! Question
+        - On the Eaton delete - do we care to keep N readings per device or device resource?  Or we are just going to delete until we have only N readings and we don't care which ones we delete?
+
+        - On the Tony delete option A, given the way we specified this, there would only be one reading in core data at all times.  Do we mean to say that we only want to keep one reading of each device or device/device resource?  Or should we first allow some arbitrary number of readings to be built up and then start deleting readings one-for-one?
+
+        - Is the configuration for core data "cache" Writable or static?
+
+### Configuration Additions
+
+No new configuration is needed to satisfy the GET operations/APIs.
+
+A new `Writable` category and set of configuration values are needed for the core data cache *delete* operation.
+
+``` toml
+[Writable]
+    [Writable.CacheMode]
+    CacheDelete="OnEach"    # or alternately could be "OnCount"
+    CacheSize=100           # specifies the number of readings core data is allowed to have before a delete is triggered.  This configuration is only used when CacheDelete="OnCount"
+    DeleteSize=10           # specifies the number of readings to remove when the cache size is exceeded.  This configuration is only used when CacheDelete="OnCount"
+```
+## Decision
+
+## Consequences
+
+One of the chief purposes of the core data cache and get latest mechanisms is to allow core command to when to query a device service and when to use existing captured sensor data.
+
+Core command will be changed so that on any GET query operation, core command will be provided with a new query parameter (`UseCache`).  When `UseCache` is specified, core command will go to core data first to retrieve the latest reading values (versus hitting the device service for the reading).  If `UseCache` is not specified, then core command will go to the device service to get the reading value.
```

Comment:
```suggestion
Core command will be changed so that on any GET query operation, core command will be provided with a new query parameter (`UseCache`).  When `UseCache` is specified, core command will go to core data first to retrieve the latest reading values (versus hitting the device service for the reading); if the reading is unavailable, core command will request it from the device service but add the flag to have it also published on the message bus.  If `UseCache` is not specified, then core command will go to the device service to get the reading value.
```

Reply:
For query parameters used by device services we use lowercase with the prefix `ds-` to indicate that these are reserved for the device service itself (vs. specific to the device service's protocol driver). For the sake of consistency, I think we should do the same and use `cs-usecache` for this parameter.

Reply:
Also did we give any consideration to making this a configuration setting vs. a per-request parameter?

Reply:
I agree with @tonyespy .  The query parameter is more flexible than the configuration.
However, I am not sure this enhancement in Core Command is required, because it's totally the same when the client invoke the Core Data API directly.